event_session/wizards/wizard_event_session.py

```diff
@@ -0,0 +1,251 @@
+        inverse_name='wizard_event_session_id',
+        string='Hours',
+    )
+    event_mail_template_id = fields.Many2one(
+        comodel_name='event.mail.template',
+        string='Mail Schedule Template',
+    )
+
+    @api.multi
+    @api.constrains('session_hour_ids')
+    def _avoid_overlapping_hours(self):
+        for hour_a in self.session_hour_ids:
+            for hour_b in self.session_hour_ids:
+                if hour_a != hour_b:
+                    if hour_a.start_time == hour_b.start_time:
+                        raise ValidationError(
+                            _("There are overlapping hours!")
+                        )
+                    elif hour_b.start_time < \
+                            hour_a.start_time < hour_b.end_time:
+                        raise ValidationError(
+                            _("There are overlapping hours!")
+                        )
+
+    @api.multi
+    def weekdays(self):
+        return (self.mondays,
+                self.tuesdays,
+                self.wednesdays,
+                self.thursdays,
+                self.fridays,
+                self.saturdays,
+                self.sundays)
+
+    @api.multi
+    def datetime_fields(self):
+        """Fields converted to Python's Datetime-based objects."""
+        result = {
+            "event_start": fields.Datetime.from_string(self.event_date_begin),
+            "event_end": fields.Datetime.from_string(self.event_date_end),
+            "day_delta": timedelta(days=1),
+        }
+        return result
+
+    @api.multi
+    def existing_sessions(self, date):
+        """Return existing sessions that match some criteria."""
+        # Todo: Improve match
+        return self.env["event.session"].search(
+            [("event_id", "=", self.event_id.id),
+             ("date", "=", date),
+             ("start_time", "=", self.start_time)],
+        )
+
+    def _get_session_mail_template(self, mail_template):
+        vals = [(6, 0, [])]
+        if isinstance(mail_template, int):
+            mail_template = self.env['event.mail.template'].browse(
+                mail_template)
+        for scheduler in mail_template.scheduler_template_ids:
+            vals.append((0, 0, {
+                'event_id': self.event_id.id,
+                'interval_nbr': scheduler.interval_nbr,
+                'interval_unit': scheduler.interval_unit,
+                'interval_type': scheduler.interval_type,
+                'template_id': scheduler.template_id.id,
+            }))
+        return vals
+
+    @api.multi
+    def create_session(self, **values):
+        """Create a new session record with the provided values."""
+        data = {
+            "name": "{} {} - {}".format(
+                strftime('%A %d/%m/%y',
+                         strptime(values["date"], "%Y-%m-%d %H:%M:%S")),
+                "%02d:%02d" % divmod(values["start_time"]*60, 60),
+                "%02d:%02d" % divmod(values["end_time"]*60, 60),
```

Comment:
I guess this works, although it is a little bit hacky to read. Once you have `date` and `lang` from above, you can do this to get it perfect:

`strftime(lang.time_format, date + datetime.timedelta(seconds=values["start_time"])`